### ISSUE TITLE
TextInput component's right prop should accept any React.ReactNode content

### ISSUE THREAD

**Issue by daileytj** [callstack/react-native-paper#2866]
**Is your feature request related to a problem? Please describe.**
I want to be able to show an `ActivityIndicator` as the `right` prop for the `TextInput` to show loading states. 

**Describe the solution you'd like**
The `TextInput` component's `right` prop should accept any `React.ReactNode` content and render it on the right side of the input.

**Describe alternatives you've considered**

**Additional context**
This is what I would expect the final result to look like
<img width="480" alt="Screen Shot 2021-08-20 at 8 44 39 AM" src="https://user-images.githubusercontent.com/13989985/130235051-05bea25c-b4d2-4e8b-89e3-6a94f1ed9890.png">

react-native: 0.65.1
react-native-paper: 4.9.2
react-native-vector-icons: 8.1.0

repro: [snack example](https://snack.expo.dev/q_-DRFxH_)

**Comment by daileytj** [comment]
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `react-native-paper` (found: `4.0.0`, latest: `4.9.2`)
> * `react-native-vector-icons` (found: `8.0.0`, latest: `8.1.0`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

yes

**Comment by daileytj** [comment]
repro provided here: [snack example](https://snack.expo.dev/q_-DRFxH_)

**Comment by TheEhsanSarshar** [comment]
@daileytj they are following material UI specs. I am not sure if material UI allows you to add any custom node on the right. correct me if I am wrong.

**Comment by shujathkhan** [comment]
@brunohkbx @daileytj If this issue is still available for grabs, I would like to contribute to it.

**Comment by brunohkbx** [comment]
@shujathkhan yes it is. You can find the docs here: https://callstack.github.io/react-native-paper/text-input-icon.html

**Comment by lukewalczak** [comment]
Hey @daileytj, the solution is not perfect, but actually you can pass your `ActivityIndicator` in that way:

```
     <TextInput
        label="Passcode *"
        value="Test"
        keyboardType={'numeric'}
        right={
          <TextInput.Affix
            text={
              <ActivityIndicator animating={true} color={'#deadbeef'} />
            }/>
        }
      />
```

**Comment by Senninseyi** [comment]
> Hey @daileytj, the solution is not perfect, but actually you can pass your `ActivityIndicator` in that way:
> 
> ```
>      <TextInput
>         label="Passcode *"
>         value="Test"
>         keyboardType={'numeric'}
>         right={
>           <TextInput.Affix
>             text={
>               <ActivityIndicator animating={true} color={'#deadbeef'} />
>             }/>
>         }
>       />
> ```

Tried this and it worked for me @lukewalczak 

then also tried this

`     <TextInput
          disabled={false}
          value="hello"
          right={
            <TextInput.Icon
              icon={() => <VerifyIcon width={20} height={20} />}
            />
          }
        />`
        
        this without any typescript error
